chore: Update to be triggered when tag pushed

LeeDongGeon1996 opened the pull request:
### Description

 - Update to be triggered when tag pushed

### Checklist

- [x] My code follows the existing style, code structure, and naming taxonomy
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my own code and included any verifying manual calculations
- [x] I have added or updated unit tests that prove my fix is effective or that my feature works, and achieves sufficient code coverage.  New and existing unit tests pass locally and in the build (below) with my changes
- [x] My changes generate no new warnings and running code analysis does not produce any issues
- [x] I have added or run the performance tests that depict optimal execution times
- [x] I have made corresponding changes to the documentation

DaveSkender commented on .github/workflows/deploy.yml:
```diff
@@ -6,6 +6,8 @@ on:
   push:
     branches:
       - main
+    tags: 
```
a pattern here might be a good preventative measure for PyPI PEP 440 compatibility, something like: `[0-9]+.[0-9]+.[0-9]+`, only for `pypi` trigger needed, since it’s the only one triggering on tags.

DaveSkender commented:
On a related note, I’ve modified the repo tag protection rules to basically prevent accidentally over-incrementing to future major version numbers, set as `[2-9]+.*`

DaveSkender commented:
Look like tag `1.1.0` auto-removed itself eventually after you delete the release note, unless you did something separately.  I didn’t.

<img width="671" alt="image" src="https://github.com/DaveSkender/Stock.Indicators.Python/assets/8432125/e75311dc-aa9f-4025-9048-8e39c4b10b91">

LeeDongGeon1996 commented:
@DaveSkender 
> On a related note, I’ve modified the repo tag protection rules to basically prevent accidentally over-incrementing to future major version numbers, set as [2-9]+.*

Thanks! And I removed the tag yesterday
